src/main/java/org/sonarqube/gradle/SonarUtils.java

```diff
@@ -99,7 +99,7 @@ static void setMainClasspathProps(Map<String, Object> properties, boolean addFor
   }
 
   static void populateJdkProperties(Map<String, Object> properties, JavaCompilerConfiguration config) {
-    properties.put("sonar.java.jdkHome", config.getJdkHome());
+    config.getJdkHome().ifPresent(s -> properties.put("sonar.java.jdkHome", s));
```

Comment:
Wow, this issue is so bad and unlucky, we should definitely have a rule targeting situation when an `Optional` is provided as argument to a method expecting an `Object` parameter... It is most likely that we would expect to have the content of the `Optional` given, and not an empty `Optional` in worst case scenario.